### PR TITLE
Fix project filter bugs

### PR DESCRIPTION
**Pull request by CaiCandong**
related: #26012

### Bugs
1. missing project filter on the issue page.
https://github.com/go-gitea/gitea/blob/1e76a824bcd71acd59cdfb2c4547806bc34b3d86/modules/indexer/issues/dboptions.go#L11-L15
3. incorrect SQL condition: some issue does not belong to a project but exists on the project_issue table. 
https://github.com/go-gitea/gitea/blob/f5dbac9d36f1678b928bee04e85fbd045c725698/models/issues/issue_search.go#L233

### Before:
before

### After:
after

### CODE REVIEW

**Review by delvh — APPROVED** [review]

File: modules/indexer/issues/dboptions.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ func ToSearchOptions(keyword string, opts *issues_model.IssuesOptions) *SearchOp
 	searchOpt := &SearchOptions{
 		Keyword:   keyword,
 		RepoIDs:   opts.RepoIDs,
+		ProjectID: &opts.ProjectID,
```
Comment (lunny) [inline modules/indexer/issues/dboptions.go]: Maybe you need to test whether project id is zero or -1?

Comment (CaiCandong) [reply modules/indexer/issues/dboptions.go]: thanks

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
But you cannot diff the `all projects` and `no project`.

**Comment by CaiCandong** [comment]
> But you cannot diff the `all projects` and `no project`.

done [08e19fd](https://github.com/go-gitea/gitea/pull/26490/commits/08e19fdad29fcc71840cdc11ba2c7ccf20ad77be)

File: modules/indexer/issues/dboptions.go [hunk 2]
```diff
@@ -17,6 +17,10 @@ func ToSearchOptions(keyword string, opts *issues_model.IssuesOptions) *SearchOp
 		IsClosed:  opts.IsClosed,
 	}
 
+	if opts.ProjectID != 0 {
+		searchOpt.ProjectID = &opts.ProjectID
+	}
+
```
Comment (yp05327) [inline modules/indexer/issues/dboptions.go:23]: if opts.ProjectID != 0
searchOpt.ProjectID = &opts.ProjectID
if opts.ProjectID == 0
searchOpt.ProjectID = &opts.ProjectID = 0

So why not add this into searchOpt directly?

Comment (CaiCandong) [reply modules/indexer/issues/dboptions.go:23]: I didn't get past the CI by doing it this way at first, so I'll change it to this and try again.

Comment (CaiCandong) [reply modules/indexer/issues/dboptions.go:23]: add this into searchOpt directly can not  past the CI

Comment (yp05327) [reply modules/indexer/issues/dboptions.go:23]: So why CI failed. I will also check this.

Comment (CaiCandong) [reply modules/indexer/issues/dboptions.go:23]: maybe about `nil` and `0`

Comment (yp05327) [reply modules/indexer/issues/dboptions.go:23]: ProjectID is a pointer, the default value is `nil` 😕

Comment (CaiCandong) [reply modules/indexer/issues/dboptions.go:23]: right, `0` is not `nil `, so we can't add it  into `searchOpt` directly

Comment (yp05327) [reply modules/indexer/issues/dboptions.go:23]: Different to other places, comes from #26012
Not bad, but need to pay attention. 😄

File: models/issues/issue_search.go [hunk 3]
```diff
@@ -168,6 +168,17 @@ func applyMilestoneCondition(sess *xorm.Session, opts *IssuesOptions) *xorm.Sess
 	return sess
 }
 
+func applyProjectCondition(sess *xorm.Session, opts *IssuesOptions) *xorm.Session {
+	if opts.ProjectID > 0 { // specific project
+		sess.Join("INNER", "project_issue", "issue.id = project_issue.issue_id").
+			And("project_issue.project_id=?", opts.ProjectID)
+	}
+	if opts.ProjectID == db.NoConditionID { // show those that are in no project
```
Comment (lunny) [inline models/issues/issue_search.go]: else if ?

**Review by lunny — APPROVED** [review]

**Pull request merged**